Does not build on GHC-8.8.2

I hope the fix is just moving parens, but I haven't tried.

Error:
```    Building library for free-categories-0.2.0.0..
    [1 of 4] Compiling Data.Quiver
    
    /private/var/folders/cn/qmlf53n918g_w0k87q4xx3rr0000gp/T/stack50980/free-categories-0.2.0.0/src/Data/Quiver.hs:139:19: error:
        • You can't specify an instance for a tuple constraint
        • In the quantified constraint ‘forall (x :: k) (y :: k).
                                        (Show (p x y), Show (q x y))’
          In the stand-alone deriving instance for
            ‘(forall x y. (Show (p x y), Show (q x y))) =>
             Show (ComposeQ p q x y)’
        |
    139 | deriving instance (forall x y. (Show (p x y), Show (q x y)))
```

--- a/src/Data/Quiver.hs
+++ b/src/Data/Quiver.hs
@@ -136,7 +136,7 @@ instance Monoid m => Category (ReflQ m) where
 
 {- | Compose quivers along matching source and target.-}
 data ComposeQ p q x z = forall y. ComposeQ (p y z) (q x y)
-deriving instance (forall x y. (Show (p x y), Show (q x y)))
+deriving instance (forall x y. Show (p x y), forall x y. Show (q x y))
   => Show (ComposeQ p q x y)
 instance (Category p, p ~ q, x ~ y)
   => Semigroup (ComposeQ p q x y) where (<>) = (>>>)
@@ -144,7 +144,7 @@ instance (Category p, p ~ q, x ~ y)
   => Monoid (ComposeQ p q x y) where mempty = id
 instance (Category p, p ~ q) => Category (ComposeQ p q) where
   id = ComposeQ id id
-  ComposeQ pyz qxy . ComposeQ pwx qvw = ComposeQ (pyz . qxy) (pwx . qvw)
+  ComposeQ yz xy . ComposeQ wx vw = ComposeQ (yz . xy) (wx . vw)
 
 {- | The left [residual]
 (https://ncatlab.org/nlab/show/residual)

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,1 +1,1 @@
-resolver: lts-14.6
+resolver: nightly-2020-01-25

